src/components/notifications/NotificationsMenu.js

```diff
@@ -149,6 +133,57 @@ function NotificationsMenu(props) {
         },
     });
 
+    const { isMarked } = useQuery({
+        queryKey: NOTIFICATIONS_MARK_ALL_SEEN_KEY,
+        queryFn: markAllSeen,
+        config: {
+            onSuccess: () => {
+                setAllNotificationsSeen();
+            },
+            onError: (error) => {
+                console.log("Error marking all notifications as Seen");
+                // *** Will add this after understanding HOC better. ***
+                // showErrorAnnouncer(
+                //     t("errorMarkAsSeen", {
+                //         count: notifications.length,
+                //     }),
+                //     error
+                // );
+            },
+        },
+    });
+
```

Comment:
2 issues:
- I don't think you can de-structure `isMarked` or a random variable from useQuery. Please refer https://react-query.tanstack.com/docs/api to find out what values are returned by the `useQuery`.
- This query is now running (although failing) any time sonora is viewed. This should only run when a user click `Mark All as Read` button. To do this, you should enable this query from `handleMarkAllAsSeenClick` function. Please refer to the `api` page I linked above more info.

Reply:
I see based on what you've commented out that you've been referencing some of Paul's code for marking a selection of notifications as seen:
https://github.com/cyverse-de/sonora/blob/9293b3c2de6f39d2ed96fa107afc2055ac10b60e/src/components/notifications/listing/index.js#L95-L109

The really big difference that you're missing (it's easy to miss, the differences are slight, and I also missed this during our Zoom chat, sorry) is that you're using `useQuery` and if you look carefully you'll see Paul is using `useMutation`.  `useQuery` is appropriate only for `GET` requests where you are only fetching or reading information.  
In Paul's case, and in your case as well, you're not just reading information, you're modifying data with a `POST` request.  For these types of requests, you must use `useMutation` which operates a little differently than `useQuery`. See docs [here](https://react-query.tanstack.com/docs/guides/mutations).

I think you'll find if you review the docs carefully and Paul's code, you'll be able to accomplish what Sriram mentioned above where this call needs to only trigger after an action (clicking on a button) occurs.

Reply:
Good catch. This should indeed use `useMutation`.

Reply:
I think you said you hadn't tested this yet, but if it doesn't work, I believe you're still missing a few refactors for switching over to `useMutation`.  The [api reference](https://react-query.tanstack.com/docs/api#usemutation) might make these differences a little more clear than the guide docs that I linked earlier.  You can also review Paul's code.